### ISSUE TITLE
How to use pyinstaller build sentence-transformers 

### ISSUE THREAD

**Issue by miilopeee** [huggingface/transformers#23636]
I want to compile sentence-transformers to executable file use pyinstaller.
This is pyinstaller .spec files:

-- mode: python ; coding: utf-8 --
from PyInstaller.utils.hooks import collect_data_files
from PyInstaller.utils.hooks import copy_metadata

import sys ; sys.setrecursionlimit(sys.getrecursionlimit() * 20)

datas = []
datas += collect_data_files('torch')
datas += copy_metadata('tokenizers')
datas += copy_metadata('sentencepiece')
datas += copy_metadata('mpmath')
datas += copy_metadata('urllib3')
datas += copy_metadata('typing-extensions')
datas += copy_metadata('tqdm')
datas += copy_metadata('threadpoolctl')
datas += copy_metadata('sympy')
datas += copy_metadata('regex')
datas += copy_metadata('pyyaml')
datas += copy_metadata('pillow')
datas += copy_metadata('packaging')
datas += copy_metadata('numpy')
datas += copy_metadata('networkx')
datas += copy_metadata('MarkupSafe')
datas += copy_metadata('joblib')
datas += copy_metadata('idna')
datas += copy_metadata('filelock')
datas += copy_metadata('click')
datas += copy_metadata('charset-normalizer')
datas += copy_metadata('certifi')
datas += copy_metadata('scipy')
datas += copy_metadata('requests')
datas += copy_metadata('nltk')
datas += copy_metadata('jinja2')
datas += copy_metadata('torch')
datas += copy_metadata('scikit-learn')
datas += copy_metadata('huggingface-hub')
datas += copy_metadata('transformers')
datas += copy_metadata('torchvision')
datas += copy_metadata('sentence_transformers')

block_cipher = None

a = Analysis(
['test.py'],
pathex=[],
binaries=[],
datas=datas,
hiddenimports=['pytorch', 'open_clip_torch'],
hookspath=[],
hooksconfig={},
runtime_hooks=[],
excludes=[],
win_no_prefer_redirects=False,
win_private_assemblies=False,
cipher=block_cipher,
noarchive=False,
)
pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

exe = EXE(
pyz,
a.scripts,
[],
exclude_binaries=True,
name='test',
debug=False,
bootloader_ignore_signals=False,
strip=False,
upx=True,
console=True,
disable_windowed_traceback=False,
argv_emulation=False,
target_arch=None,
codesign_identity=None,
entitlements_file=None,
)
coll = COLLECT(
exe,
a.binaries,
a.zipfiles,
a.datas,
strip=False,
upx=True,
upx_exclude=[],
name='test',
)

When I compile successfully, run the exe,the exe output:
Segmentation fault: 11.

How should I solve this problem？thx
